src/libs/Tiktoken.Core/CoreBPE.cs

```diff
@@ -86,12 +87,12 @@ public int CountTokensNative(string text)
         var tokens = 0;
 #if NET7_0_OR_GREATER
         var textSpan = text.AsSpan();
+        Span<byte> pieceBytes = stackalloc byte[128];
```

Comment:
> Since UTF-8 encoding can require up to 4 bytes per UTF-16 character

This is false. Where are you getting this info, Coderabbit?

Reply:
I think it would be very uncommon to have a match longer than 42 characters. But if there were, it would just allocate a temporary buffer to decode into.